pkg/restore/restore.go

```diff
@@ -642,7 +643,19 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 		ctx.infof("Restoring %s: %v", obj.GroupVersionKind().Kind, obj.GetName())
 		_, err = resourceClient.Create(obj)
 		if apierrors.IsAlreadyExists(err) {
-			addToResult(&warnings, namespace, err)
+			if obj.GroupVersionKind().Kind == "ServiceAccount" {
+				fromCluster, err := resourceClient.Get(obj.GetName(), metav1.GetOptions{})
+				desired, err := mergeSecrets(fromCluster, obj)
+				desiredBytes, err := json.Marshal(desired)
+				fromClusterBytes, err := json.Marshal(obj)
+				patchBytes, err := jsonmergepatch.CreateThreeWayJSONMergePatch(fromClusterBytes, desiredBytes, nil)
```

Comment:
if `mergeSecrets` always returns a non-nil value as I recommended, then it's possible the patch will be empty (i.e. there's no diff between `desired` and `fromCluster`). We should check for that case and not call `Patch()` if so.